api/base/parsers.py

```diff
@@ -15,16 +16,18 @@ def parse(self, stream, media_type=None, parser_context=None):
         Parses the incoming bytestream as JSON and returns the resulting data
         """
         result = super(JSONAPIParser, self).parse(stream, media_type=media_type, parser_context=parser_context)
+        if not isinstance(result, dict):
+            raise ValidationError("Invalid data. Expected a dictionary but got {}".format(type(result)))
```

Comment:
Okay, prior to the formatting changes we returned this: 
![screen shot 2015-09-24 at 2 44 35 pm](https://cloud.githubusercontent.com/assets/9755598/10083081/ca4ae03c-62ca-11e5-9ff6-08a8d8891e90.png)